ts/Core/Chart/Chart.ts

```diff
@@ -1205,8 +1205,20 @@ class Chart {
         }
 
         if (options && !elem) {
+            // If title text is unset, and not explicitly unset,
+            // default to lang option
+            const text = options.text ?? (
+                key === 'title' &&
+                    // If the title option is set to null,
+                    // it will be an empty object at this point
+                    Object.keys(options).length &&
+                    !('text' in options) ?
+                    chart.options.lang.chartTitle :
+                    void 0
+            );
+
```

Comment:
Rethinking this, I _think_ we can avoid this whole complex condition by just adding this at the end of the `setOptions` function:

```ts
    if (options.lang?.chartTitle) {
        defaultOptions.title.text = options.lang.chartTitle;
    }
```

It would be shorter, but also way easier to grasp and to maintain.

Unfortunately the `yAxis.title.text` doesn't seem to be that simple, because the default y-axis options are a bit buried. But it may be possible also there, by importing the appropriate object into the Defaults.ts.